### DISCUSSION TITLE
大厂开源之殇 | 夜天之书

### DISCUSSION THREAD

**Discussion by giscus** [tisonkun/blog#76] [Comments]
# 大厂开源之殇 | 夜天之书

本轮开源之风吹起迄今数年，最大的影响还是越来越多的商业公司开始探索开源方法能够如何改变自己的经营策略。 开源策略循序渐进分成使用、参与和发起。在发起开源项目实践一线的，一个是打着开源旗号的创业公司，另一个就是大型企业尤其互联网企业（大厂）。 前者我在过去的文章里已经详细讨论了他们的动机、面临的风险和应对策略。 本文将关注大厂发起的开源项目的共同特点，讨论围绕这样的开源项目聚拢起来的企业研发团队、市

https://www.tisonkun.org/2023/04/13/open-source-mismatching/

**Comment by ZoomQuiet** [comment]
#是也乎,(￣▽￣)

> ..工程效能团队

这听起来就像大厂里针对工程师的西厂...也难为这一职能部门了, 想在大家已经福报的压力中,
进一步要求公开效能数据, 以及进一步优化, 实在太挑战人性了...

其实大厂这些开源走形的根本原因, 并不是不知道这么作有问题, 
而是 "上游思想" 的正确反应吧;

- 因为, 自己是绝对的上游
- 同时, 自己又是绝对的第一下游
- 那么, 其它次下游的用户是否用的起来, 是否能反馈足够品质的代码, 并不重要

设立"开发者关系团队"其实, 不过是 fork 出又一个类似 工程效能团队 的背锅侠团队,
毕竟这个团队也是有 KPI 的, 如何定能让这个团队的绩效每年有合理增长?

最近倒是发现一个新的思路:
[简介 - Serverless Devs](https://docs.serverless-devs.com/serverless-devs/yaml_and_cli)

- AliYUN 直接将原先 FC/函数计算的配套本地调试工具给砍了
- "外包" 给了一个通用函数计算支持工具 `s\` ~ 也就是这个 Serverless Devs
- 这是一个独立社区, 而且不仅仅支持 AliYUN 的函数计算, 同时也支持其它云厂商各种微服务的日常开发者生命周期
- 其中当然有 ALiYUN 的工程师, 但是, 是否算日常工作另外说, 不过, 这样等于直接将一个核心工具分离为独立运营/运维/宣传的开源社区, 和内部流程分离, 气味应该更加纯粹些?

**Reply by tisonkun** [reply]
> 开发者关系

并不只是为开源服务。一般而言，用户的 awareness 和 adoption 是主要的指标。

企业对开源做投入，总得有人负责。OSPO 是大厂处理多个开源项目时候的安全、合规和治理问题。开发者关系是处理单个项目产品从内部研发到外部用户开发者的沟通。

> 那么

绝对的上游对于企业开源来说不好改变，除非是行业竞争大，需要组成联盟，即 Java / Rust 或 Kubernetes / Istio 的情况，这个在国内的实践还很少。绝对的下游是可以改变的，文中有成功的例子，取决于项目怎么发展起来，用户的 adoption 和开发者的参与对项目的价值如何被衡量。这个认知在企业内是可以改变的。

> 气味应该更加纯粹些

不知道具体的组织架构。BeyondStorage 就是青云一次【兼容所有对象服务】的开源实验，以失败告终。
